scripts/linters/pylint_extensions.py

```diff
@@ -1959,6 +2028,382 @@ def process_tokens(self, tokens):
+            if token == 'def':
+                outside_function_signature_block = False
+                func_def_start_line = line_num
+                outside_args_section = False
+
+        if token_type == tokenize.OP:
+            if token == '->':
+                outside_args_section = True
+                args_section_end_line_num = line_num
+            if outside_args_section and token == ':':
+                outside_function_signature_block = True
+
+        # Checking if exceptional_type is present in function definition or not.
+        if token_type == tokenize.NAME and token == exceptional_type:
+            if not outside_args_section:
+                type_present_inside_arg_section = True
+            elif (
+                outside_args_section and
+                args_section_end_line_num == line_num
+            ):
+                type_present_inside_return_section = True
+
+        if (
+            type_present_inside_arg_section or
+            type_present_inside_return_section
+        ):
+            type_present_in_function_signature = True
+
+        if outside_function_signature_block:
+            if type_present_in_function_signature:
+                if (
+                    type_comment_present and
+                    func_def_start_line <= (
+                        type_comment_line_num +
+                        self.ALLOWED_LINES_OF_GAP
+                    )
+                ):
+                    type_comment_present = False
+                else:
+                    if exceptional_type == 'Any':
+                        self.add_message(
+                            'any-type-used', line=func_def_start_line)
+                    if exceptional_type == 'object':
+                        self.add_message(
+                            'object-class-used', line=func_def_start_line)
+
+                type_present_in_function_signature = False
+                type_present_inside_arg_section = False
+                type_present_inside_return_section = False
+            if token_type == tokenize.NAME and token == exceptional_type:
+                if exceptional_type == 'object':
+                    # Excluding the case when object is called:
+                    # Eg: var = object()
+                    if 'object()' in line:
+                        return
+                # Passing those cases where Any is imported.
+                if exceptional_type == 'Any':
+                    if (
+                        import_status_dict['single_line_import'] and
+                        import_status_dict['import_line_num'] == line_num
+                    ):
+                        return
+                    elif import_status_dict['inside_multi_line_import_scope']:
+                        return
+                # Checking if comment for exceptional_type is present and it's
+                # with in the range (minimum 10 line of gaps).
+                if (
```

Comment:
Isn't this statement a repetition from above?
should we move it to a helper function?

Reply:
If we see the `if clause` there we have different conditions but yeah both of the `else clauses` have similar code so i pushed it into the helper function PTAL!

Reply:
this if clause can be a helper function right?

```py
def _comment_present_within_scope(self, line_number, comment_line_number):
    return line_number <= comment_line_number + ALLOWED_LINES_OF_GAP_IN_COMMENT
```

Something like this, so that all the conditions can use this?
Feel free to change the helper function's name to something meaningful

Reply:
I have some thoughts on this:
1. Passing variables just for 1 line of code does not sound good to me.
2, In the future, If we need to change the logic for any of the exceptional types ( cast, any, and object) then we have to take down this function again.